pipeline-modules/app-service/azure/git-clone.yaml

```diff
@@ -0,0 +1,36 @@
+provisioner: azure
+name: git-clone
+version: 1
+revision: 1
+displayName: Git clone
+description: Clone remote Git repository and checkout a given commit if specified.
+target: ""
+keywords:
+  - git
+  - clone
+  - checkout
+author: CloudCover
+meta: {}
+inputs:
+  properties:
+    git_checkout_target:
+      title: Checkout Target
+      description: Git branch, tag or commit hash to checkout.
+      type: string
+      default: ""
+    git_local_path:
+      title: Local Clone Directory
+      description: The name of a new directory to clone into.
+      type: string
+      default: repo
+    git_remote_path:
+      title: Repository URL
+      description: The repository to clone from.
+      type: string
+  required:
+    - git_remote_path
+template: |
+  steps:
+  - script: |
+      git clone {% if git_checkout_target %}--no-checkout{% endif %} https://{{ git_remote_path }} {{ git_local_path }}{% if git_checkout_target %}
+      cd {{ git_local_path }} && git checkout {{ git_checkout_target }} && cd .. {% endif %}
```

Comment:
can we put this in workspace dir?
In order for other app integration pipelines to run scans, test, and build. the location should already be present in a variable.
for eg git clone https://.../../xyz-project will create a folder named `xyz-project`, so we need to split the URL by ''/" pick the last element of split array and add it as a workspace folder in a variable.


This can be done as separate task as well